### ISSUE TITLE
Some results are not reproducible

### ISSUE THREAD

**Issue by fake-warrior8** [haojc/ShufflingVideosForTSG#8]
Hi, I used the given commands to train your model and the baseline. However, some results are not reproducible. I used your code, bbox features and glove word embedding for the Anet dataset. These results are the test result of the valid set best mIoU  epoch (as you said before).

![1](https://user-images.githubusercontent.com/40395156/218364124-31192ad0-74a5-495f-ae38-7fc16ba56726.png)

1. The baseline result on the Charades dataset are too bad.
2. On the Anet dataset, the paper's method has little improvement over the baseline.
3.  I ran the paper's method twice on the Charades dataset, the results fluctuate a lot.

**Comment by haojc** [comment]
Thanks for pointing out the problems. I am re-running these experiments to verify the results you obtained. I'll contact you once I finish them. And I can answer some of the questions now.
For Q1: 
Yes, the training processing of the baseline on Charades is not very steady. Sometimes the model gets degradation and performs poorly. We also find that applying our training framework can effectively address this training problem.
If the model is trained normally, the results are around the ones we reported in the paper. 
For Q2&3:
I'll contact you once I finish re-running them. 
I have finished two runs on Charades, and the results are  44.48   76.15  64.77   46.52   27.26   6.07; 43.76   77.07  63.94   44.74   25.81   5.75, respectively. I also advise you to perform more runs if you would like. Btw, you can also directly select 16th epoch's model as the best one, which empirically shows the best performance on test-ood set.

**Comment by fake-warrior8** [comment]
> Thanks for pointing out the problems. I am re-running these experiments to verify the results you obtained. I'll contact you once I finish them. And I can answer some of the questions now. For Q1: Yes, the training processing of the baseline on Charades is not very steady. Sometimes the model gets degradation and performs poorly. We also find that applying our training framework can effectively address this training problem. If the model is trained normally, the results are around the ones we reported in the paper. For Q2&3: I'll contact you once I finish re-running them. I have finished two runs on Charades, and the results are 44.48 76.15 64.77 46.52 27.26 6.07; 43.76 77.07 63.94 44.74 25.81 5.75, respectively. I also advise you to perform more runs if you would like. Btw, you can also directly select 16th epoch's model as the best one, which empirically shows the best performance on test-ood set.

I ran all the experiments three times, the three results above still show.
In addition, I found that the results of the best test set picked from the best validation set did not differ much from the best test set picked directly (mIoU difference < 0.5), which implies that choosing the 16th epoch may also be ineffective.![TGC0NWIR@Z1VO~4Q`M@9ZLQ](https://user-images.githubusercontent.com/40395156/218722796-4bfaf34c-246c-4b1b-9b90-014b990943ba.png)

**Comment by haojc** [comment]
I have finished the experiments re-runs and the results are shown as follows. 
![image](https://user-images.githubusercontent.com/30310473/219258703-d31f4530-cfbb-44cc-bcb1-04cebc9c6f19.png)
When I released this version project on Github, I didn't just copy the codes from the draft project but re-write them to achieve better readability. After that, I only checked the results of ShuffleVideo and released two pre-trained models. But there seems to be some problem with the baseline method. So I performed careful re-runs for both ShuffleVideo and baseline.
The summarized points according to my experiment results:
1. The performance of baseline and ShuffleVideos on Charades, and ShuffleVideo on ANet are almost consistent with the results in the paper. (For baseline on Charades, here I change the optimizer from Adam to Adamw to achieve more steady training) As for all of your results of ShuffleVideo on Charades are lower, I think there may be two potential causes. First is randomness. For each video in each training epoch, we randomly shuffle the target moment to another position. I will consider adding random seeds to ensure similar results. Second is that are you sure your hyper-parameter and environment setting are the same as the ones we provided?
2. The performance of baseline on Anet by this version code is indeed higher than the ones in my past experiments (blue colored in the table). I will check the reason later.
3. The epoch affects the grounding performance a lot. Empirically, the more training epoch, the higher performance achieved on higher IoU thresholds (IoU=0.5, 0.7). The following shows the results picked from the best test set.
![image](https://user-images.githubusercontent.com/30310473/219266291-0ce85bad-c358-4319-9be0-4100f9a33212.png)

**Comment by fake-warrior8** [comment]
> I have finished the experiments re-runs and the results are shown as follows. ![image](https://user-images.githubusercontent.com/30310473/219258703-d31f4530-cfbb-44cc-bcb1-04cebc9c6f19.png) When I released this version project on Github, I didn't just copy the codes from the draft project but re-write them to achieve better readability. After that, I only checked the results of ShuffleVideo and released two pre-trained models. But there seems to be some problem with the baseline method. So I performed careful re-runs for both ShuffleVideo and baseline. The summarized points according to my experiment results:
> 
> 1. The performance of baseline and ShuffleVideos on Charades, and ShuffleVideo on ANet are almost consistent with the results in the paper. (For baseline on Charades, here I change the optimizer from Adam to Adamw to achieve more steady training) As for all of your results of ShuffleVideo on Charades are lower, I think there may be two potential causes. First is randomness. For each video in each training epoch, we randomly shuffle the target moment to another position. I will consider adding random seeds to ensure similar results. Second is that are you sure your hyper-parameter and environment setting are the same as the ones we provided?
> 2. The performance of baseline on Anet by this version code is indeed higher than the ones in my past experiments (blue colored in the table). I will check the reason later.
> 3. The epoch affects the grounding performance a lot. Empirically, the more training epoch, the higher performance achieved on higher IoU thresholds (IoU=0.5, 0.7). The following shows the results picked from the best test set.
>    ![image](https://user-images.githubusercontent.com/30310473/219266291-0ce85bad-c358-4319-9be0-4100f9a33212.png)

My hyper-parameters are the same as the ones you provided.   My environment is a little different with yours. I ran all code in a pytorch1.12 conda env. Overall, the reimplementation results are acceptable.
